pages/project-version/middleware/index.js

```diff
@@ -7,6 +9,110 @@ const getVersion = () => (req, res, next) => {
     .catch(next);
 };
 
+const isNew = (user, activityLog, createdAt) => {
+  const activity = user.profile.asruUser ? /returned-to-applicant$/ : /resubmitted$/;
+  const logItem = activityLog.find(e => e.eventName.match(activity));
+  return !logItem || createdAt > logItem.createdAt;
+};
+
+const getComments = () => (req, res, next) => {
+  if (!req.project.openTasks.length) {
+    return next();
+  }
+  req.api(`/tasks/${req.project.openTasks[0].id}`)
+    .then(({ json: { data } }) => {
+      const status = req.user.profile.asruUser ? 'returned-to-applicant' : 'resubmitted';
+      const statuses = data.activityLog.filter(e => e.eventName.match(/^status:/));
+      // if most recent status change is returned to applicant (asru user) or
+      // resubmitted (non-asru), ignore comments made after this status change.
+      const ignoreAfter = statuses[0] && statuses[0].event.status === status && statuses[0].createdAt;
+      const comments = chain(data.activityLog)
+        .filter(e => e.eventName === 'comment')
+        .filter(e => !ignoreAfter || e.createdAt < ignoreAfter)
+        .groupBy(comment => comment.event.meta.payload.meta.field)
+        .mapValues(comments =>
+          comments.map(({ comment, createdAt, changedBy: { firstName, lastName } }) => {
+            return {
+              comment,
+              // we want to show the date of the following status change, not the comment submission.
+              createdAt: ([...statuses].reverse().find(s => s.createdAt > createdAt) || {}).createdAt,
+              author: `${firstName} ${lastName}`,
+              isNew: isNew(req.user, data.activityLog, createdAt)
+            };
+          })
+            .reverse()
```

Comment:
Is there a reason this is not part of the chain? It's a bit odd there being one call which is one level lower.

Reply:
yes

Reply:
the chain at this point is an object, the values are the arrays which need reversing